internal/cli/run/run.go

```diff
@@ -112,17 +110,34 @@ func init() {
 				return err
 			}
 		}
+		return nil
+	})
 
-		if *nettestGroup == "" {
-			log.Infof("Running %s tests", color.BlueString("all"))
-			for tg := range groups.NettestGroups {
-				if err := runNettestGroup(tg, ctx, network); err != nil {
-					log.WithError(err).Errorf("failed to run %s", tg)
-				}
+	websitesCmd := cmd.Command("websites", "")
+	websitesCmd.Action(func(_ *kingpin.ParseContext) error {
+		return runNettestGroup("websites", ctx, network)
+	})
+	imCmd := cmd.Command("im", "")
+	imCmd.Action(func(_ *kingpin.ParseContext) error {
+		log.Info("calling im code path")
+		return runNettestGroup("im", ctx, network)
+	})
+	performanceCmd := cmd.Command("performance", "")
+	performanceCmd.Action(func(_ *kingpin.ParseContext) error {
+		return runNettestGroup("performance", ctx, network)
+	})
+	middleboxCmd := cmd.Command("middlebox", "")
+	middleboxCmd.Action(func(_ *kingpin.ParseContext) error {
+		return runNettestGroup("middlebox", ctx, network)
+	})
+	allCmd := cmd.Command("all", "").Default()
+	allCmd.Action(func(_ *kingpin.ParseContext) error {
+		log.Infof("Running %s tests", color.BlueString("all"))
+		for tg := range groups.NettestGroups {
+			if err := runNettestGroup(tg, ctx, network); err != nil {
+				log.WithError(err).Errorf("failed to run %s", tg)
```

Comment:
Do we want this to happen? Would we rather want the probe to continue and then eventually `exit(1)`. This seems to me a problem similar to `make` vs `make -k`. I'd say we discuss this now, then this code is fine anyway, and if we decide that the default is to continue, we file an issue for later. (I'd rather not introduce a `-k`, but maybe you disagree?)

Reply:
I am not sure I understand entirely. Do you mean why don't we instead exit when the first test of a group fails?

I don't think that would be desirable as we still wait to run the remaining tests inside of the group even if a single instance of it is to fail.

Maybe it would be wise to in some way record this and return a different exit code when the process terminates later on to indicate the failure, but I don't think we should be exiting early here.

Reply:
Ah, reading more carefully, `Errorf` does not exit. This is what I'd like to happen.

Reply:
😕 https://godoc.org/github.com/apex/log#Entry.Errorf